src/main/java/io/vertx/mqtt/MqttClientOptions.java

```diff
@@ -74,7 +74,7 @@ public MqttClientOptions() {
    */
   public MqttClientOptions(JsonObject json) {
     super(json);
-    //TODO implement this constructor
+    MqttClientOptionsConverter.fromJson(json, this);
```

Comment:
What is _MqttClientOptionsConverter_ ? Where is the class located?

Reply:
It's been automatically generated with the help of the `@DataObject(generateConverter=true)` annotation.

As it's generated as part of the build, I've not that class under revision control.
If you're interested, generated source is being placed in `src/generated`.

Reply:
Ok, great. Thank you for details.

Reply:
@sauthieg you can push this file in the PR don't worry.

Reply:
Well, as it's being generated, I would prefer to keep it outside of rev control, otherwise for any new configuration attribute, we'll have to remember to copy the (changed) generated file in the sources.

Reply:
Understood. @vietj what do you think about that ? For example in the Vert.x Kafka client we already have a generated file pushed to the repo (https://github.com/vert-x3/vertx-kafka-client/blob/master/src/main/generated/io/vertx/kafka/client/common/PartitionInfoConverter.java)

Reply:
I think it's a misconfiguration issue in the project, it should be committed in the repo.

Reply:
Ok, but we keep it in the `src/main/generated` directory ?
At least, git will show delta if something change in this file

Reply:
Yes, we already have it in the Kakfa client repo.

Reply:
it is actually wished to be committed as the source code depends on it, it allows someone that clones the project to find the source code already there.

Reply:
updated